### PR TITLE
RST: update highlights

### PR DESCRIPTION
**Pull request by stsewd**

### CODE REVIEW

File: queries/rst/highlights.scm [hunk 1]
```diff
@@ -20,7 +20,14 @@
   name: (type) @include)
  (#match? @include "^include::$"))
 
-; Blocks
+((directive
+   name: (type) @function.builtin)
+ (#match?
+  @function.builtin
+  ; https://docutils.sourceforge.io/docs/ref/rst/directives.html
+  "^(attention|caution|danger|error|hint|important|note|tip|warning|admonition)|(image|figure)|(topic|sidebar|line-block|parsed-literal|code|math|rubric|epigraph|highlights|pull-quote|compound|container)|(table|csv-table|list-table)|(contents|sectnum|section-numbering|header|footer)|(target-notes)|(meta)|(replace|unicode|date)|(raw|class|role|default-role|title|restructuredtext-test-directive)::$"))
```
Comment (stsewd) [inline queries/rst/highlights.scm:28]: Not sure if there is a way to split this loooooooong line

File: queries/rst/highlights.scm [hunk 2]
```diff
@@ -58,8 +65,30 @@
...
+
+((interpreted_text
+  (role) @_role
+  "interpreted_text" @text.strong)
+ (#eq? @_role ":strong:"))
+
+; Sufix role
+((interpreted_text
+  "interpreted_text" @text.emphasis
+  (role) @_role)
```
Comment (stsewd) [inline queries/rst/highlights.scm:82]: Not sure if there is a convention for this, capturing a node, but not using it for the query itself.

File: queries/rst/highlights.scm (hunk 1 above)
Comment (theHamsta) [reply queries/rst/highlights.scm:28]: for me it's fine. There are similar monsters for other languages, also in the upstream queries.

File: queries/rst/highlights.scm (hunk 2 above)
Comment (theHamsta) [reply queries/rst/highlights.scm:82]: why not leave `@_role` away then?

File: queries/rst/highlights.scm [hunk 3]
```diff
@@ -56,10 +63,32 @@
...
+ (#eq? @_role ":emphasis:"))
+
+((interpreted_text
+  "interpreted_text" @text.strong
```
Comment (theHamsta) [inline queries/rst/highlights.scm:86]: Is this really verbatim "interpreted_text"?

Comment (stsewd) [reply queries/rst/highlights.scm:86]: Yeah, it's an anonymous node

**Pull request merged**